core/templates/cowdata.h

```diff
@@ -269,22 +269,30 @@ void CowData<T>::_unref() {
 
 	SafeNumeric<USize> *refc = _get_refcount();
 	if (refc->decrement() > 0) {
-		return; // still in use
+		// Data are still in use elsewhere.
+		_ptr = nullptr;
+		return;
 	}
-	// clean up
+	// Clean up.
+	// First, invalidate our own reference.
+	// It is required to do so immediately because it must not be observable
+	//  outside of this function after refcount has already been reduced to 0.
+	// It must be done before calling the destructors, because one of them may
+	//  otherwise observe it through a reference to us. In this case, it may try
+	//  to access the buffer, which is illegal after some of the elements in it
+	//  have already been destructed, and may lead to a segmentation fault.
```

Comment:
Going off our comment style guidelines[^1], you can separate the multi-line sections with admonitions & utilize a width of ~100 characters rather than ~80
```suggestion
	// Clean up.
	// First, invalidate our own reference.
	// NOTE: It is required to do so immediately because it must not be observable outside of this
	// function after refcount has already been reduced to 0.
	// WARNING: It must be done before calling the destructors, because one of them may otherwise
	// observe it through a reference to us. In this case, it may try to access the buffer, which
	// is illegal after some of the elements in it have already been destructed, and may lead to
	// a segmentation fault.
```
[^1]:https://docs.godotengine.org/en/latest/contributing/development/code_style_guidelines.html#comment-style-guide